src/Compiler/CMakeLists.txt

```diff
@@ -1,25 +1,25 @@
 # SPDX-License-Identifier: Apache-2.0
 
-# CMAKE_CFG_INTDIR is . for single-config generators such as make, and
-# it has a value (e.g. $(Configuration)) otherwise, so we can use it to
-# determine whether we are dealing with a multi-config generator.
-if (NOT "${CMAKE_CFG_INTDIR}" STREQUAL ".")
-  set(FILE_GENERATE_DIR ${CMAKE_CURRENT_BINARY_DIR}/${CMAKE_CFG_INTDIR})
-else()
-  set(FILE_GENERATE_DIR ${CMAKE_CURRENT_BINARY_DIR}/${CMAKE_BUILD_TYPE})
-endif()
-
 configure_file(
   ${CMAKE_CURRENT_SOURCE_DIR}/ExternalUtil.hpp.in
   ${CMAKE_CURRENT_BINARY_DIR}/ExternalUtil.hpp.cfg
   @ONLY
   )
 
 file(GENERATE
-  OUTPUT ${FILE_GENERATE_DIR}/ExternalUtil.hpp
+  OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/$<CONFIG>/ExternalUtil.hpp
   INPUT ${CMAKE_CURRENT_BINARY_DIR}/ExternalUtil.hpp.cfg
   )
 
+# CMAKE_CFG_INTDIR is . for single-config generators such as make, and
+# it has a value (e.g. $(Configuration)) otherwise, so we can use it to
+# determine whether we are dealing with a multi-config generator.
+if (NOT "${CMAKE_CFG_INTDIR}" STREQUAL ".")
+  set(FILE_GENERATE_DIR ${CMAKE_CURRENT_BINARY_DIR}/${CMAKE_CFG_INTDIR})
+else()
+  set(FILE_GENERATE_DIR ${CMAKE_CURRENT_BINARY_DIR}/${CMAKE_BUILD_TYPE})
+endif()
+
 add_custom_target(ExternalUtil DEPENDS ${FILE_GENERATE_DIR}/ExternalUtil.hpp)
 
```

Comment:
This should also be changed to `${CMAKE_CURRENT_BINARY_DIR}/$<CONFIG>/ExternalUtil.hpp`?

Reply:
No, the targets get evaluated at build time when the configuration is known, so it shouldn't use a generator expression. The `file` call gets evaluated at cmake configuration time, so it needs to use the generator expression.

Reply:
Oh right. You set `FILE_GENERATOR_DIR` above. Sorry, for some reason I can never read the github diff on the web page properly.